doc/qos/tunnel_dscp_remapping.md

```diff
@@ -0,0 +1,238 @@
+# DSCP remapping for tunnel traffic
+
+## 1 Table of Content ###
+
+- [Revision](#11-revision)
+- [Scope](#2-scope)
+- [Definitions/Abbreviations](#3-definitionsabbreviations)
+- [Overview](#4-overview)
+- [Design](#5-design)
+    - [SWSS Schema](#51-swss-schema)
+        - [Define new table for mapping](#511-define-new-table-for-mapping)
+        - [Update existing TUNNEL table](#512-update-existing-tunnel-table)
+        - [Define new field for extra lossless queues](#513-define-new-field-for-extra-lossless-queues)
+    - [SAI attribute](#52-sai-attribute)
+    - [orchagent](#53-orchagent)
+- [Test requirement](#6-test-requirement)
+- [Open Questions](#7-open-questions)
+
+### 1.1 Revision ###
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Bing Wang   | Initial version                   |
+
+
+## 2 Scope ##
+
+This document covers high level design of DSCP and TC remapping for tunnel traffic in SONiC.
+
+## 3 Definitions/Abbreviations ##
+
+
+| Term | Meaning |
+|:--------:|:---------------------------------------------:|
+| PFC | Priority-based Flow Control  |
+| TC | Traffic class|
+| DSCP| Differentiated Services Code Point |
+
+## 4 Overview
+
+In Dual-ToR scenario, PFC deadlock can happen if two servers are congested at same time and start sending PFC pause frames to both upper ToR and lower ToR. It is because the south bound traffic from T1 to standby ToR is bounced back to T1 via the same queue. Even when the block condition is resolved, the pause condition among T1 and both ToRs can stay forever and no traffic will go through.
+
+To avoid this scenario, the bounced-back traffic from standby ToR to T1 will be remapped into another queue. For example, the traffic flow is as below if we are going to remap traffic in queue 3 to queue 2.
+
+<p align=center>
+<img src="dscp-remapping-images/Bounced-back-traffic-flow.png" alt="Figure 1. Bounced back traffic flow">
+</p>
+
+When congestion happens on server, the traffic flow of PFC pause frames is as below diagram.
+
+<p align=center>
+<img src="dscp-remapping-images/Bounced-back-traffic-flow-PFC.png" alt="Figure 2. Bounced back traffic flow with PFC pause">
+</p>
```

Comment:
in order to send PFC to priority 2, should `PFC_PRIORITY_TO_PRIORITY_GROUP_MAP` be updated as well? currently only [3 and 4](https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/qos_config.j2#L188) is enabled.

Reply:
Only some vendors use that mapping.